parachain/pallets/ethereum-beacon-client/src/tests.rs

```diff
@@ -418,6 +417,27 @@ fn submit_update_with_sync_committee_in_current_period() {
 	});
 }
 
+#[test]
+fn submit_update_in_next_period() {
+	let checkpoint = load_checkpoint_update_fixture();
+	let sync_committee_update = load_sync_committee_update_fixture();
+	let update = load_next_finalized_header_update_fixture();
+	let sync_committee_period = compute_period(sync_committee_update.finalized_header.slot);
+	let next_sync_committee_period = compute_period(update.finalized_header.slot);
+	assert_eq!(sync_committee_period + 1, next_sync_committee_period);
+
+	new_tester().execute_with(|| {
+		assert_ok!(EthereumBeaconClient::process_checkpoint_update(&checkpoint));
+		assert_ok!(EthereumBeaconClient::submit(
+			RuntimeOrigin::signed(1),
+			sync_committee_update.clone()
+		));
+		assert_ok!(EthereumBeaconClient::submit(RuntimeOrigin::signed(1), update.clone()));
+		let block_root: H256 = update.finalized_header.clone().hash_tree_root().unwrap();
+		assert!(<FinalizedBeaconState<Test>>::contains_key(block_root));
+	});
+}
+
```

Comment:
Accidentally removed this test in https://github.com/Snowfork/snowbridge/pull/894/files. Adding it back.